components/PaginationLinks.js

```diff
@@ -1,13 +1,15 @@
 import Link from 'next/link';
+import {useRouter} from "next/router";
 
-const PaginationLinks = ({ offset, username, hideOnFirst }) => {
-  const normalOffset = +(offset || 0); // normalize offset which can be empty string or undefined
+const PaginationLinks = ({ hideOnFirst, pathname}) => {
+  const { query: { offset } } = useRouter();
 
+  const normalOffset = +(offset || 0); // normalize offset which can be empty string or undefined
   const newerOffset = Math.max(normalOffset - 30, 0);
   const olderOffset = normalOffset + 30;
 
-  const newerLink = '/' + username + (newerOffset ? '?offset=' + newerOffset : '');
-  const olderLink = '/' + username + '?offset=' + olderOffset;
+  const newerLink = pathname + (newerOffset ? '?offset=' + newerOffset : '');
+  const olderLink = pathname + '?offset=' + olderOffset;
```

Comment:
This won't work if `pathname` has parameters (and `?`) in it, like in Search. But I guess we can leave that fix for future updates, this is good enough for HomePage and UserPage.